engine/src/main/java/org/hibernate/search/engine/search/projection/dsl/FieldProjectionValueStep.java

```diff
@@ -25,11 +25,27 @@ public interface FieldProjectionValueStep<N extends FieldProjectionOptionsStep<?
 	/**
 	 * Defines the projection as multi-valued, i.e. returning {@code List<T>} instead of {@code T}.
 	 * <p>
-	 * Calling {@link #multi()} is mandatory for multi-valued fields,
+	 * Calling {@link #multi()}/{@link #multi(MultiProjectionTypeReference)} is mandatory for multi-valued fields,
 	 * otherwise the projection will throw an exception upon creating the query.
 	 *
 	 * @return A new step to define optional parameters for the multi-valued projections.
 	 */
-	FieldProjectionOptionsStep<?, List<T>> multi();
+	default FieldProjectionOptionsStep<?, List<T>> multi() {
+		return multi( MultiProjectionTypeReference.list() );
+	}
+
+	/**
+	 * Defines the projection as multi-valued, i.e. returning a collection, e.g. {@code List<T>}, instead of {@code T}.
+	 * <p>
+	 * Calling {@link #multi()}/{@link #multi(MultiProjectionTypeReference)} is mandatory for multi-valued fields,
+	 * otherwise the projection will throw an exception upon creating the query.
+	 * <p>
+	 * Requires a collection type reference, either a built-in (see {@link MultiProjectionTypeReference}) or a custom one.
+	 *
+	 * @param collectionTypeReference Collection type reference that specifies the expected collection into which the values have to be collected into.
+	 * @return A new step to define optional parameters for the multi-valued projections.
+	 * @see MultiProjectionTypeReference
+	 */
+	<C> FieldProjectionOptionsStep<?, C> multi(MultiProjectionTypeReference<C, T> collectionTypeReference);
```

Comment:
I wonder whether renaming this from multi to container and then adding built-in type references for optional is a good idea ...

Reply:
+1 for `container` as the default is already "multi", so it feels a bit weird to introduce a "multi".

However, don't we expose "multi" methods in other DSLs, especially in the search DSL? We'd need to be consistent...

Reply:
yeah we also have that `HighlightProjectionOptionsStep#single()` ... 
It would be nice if we can just deprecate all that and do something like

```java
.select(
	f -> f.composite()
		.from(
			f.highlight( "anotherString" ),  // default list as it is now
			f.highlight( "anotherString" ).accumulator(ProjectionAccumulator.single() ), // instead of calling  f.highlight( "anotherString" ).single(),
			f.highlight( "anotherString" ).accumulator(ProjectionAccumulator.optional() ), // still single but wraps the string into an optional
			f.highlight( "anotherString" ).accumulator(ProjectionAccumulator.sortedSet() ), // multi but non-default collection type
			f.highlight( "anotherString" ).accumulator(ProjectionAccumulator.sortedSet(customComparator) ), // multi but non-default collection type with custom comparator
			f.highlight( "anotherString" ).accumulator(ProjectionAccumulator.array() ), // multi but non-default collection type
			f.highlight( "anotherString" ).accumulator(CustomUserDefinedThing.smth() ), // whatever the user came up with
		).asList()
)
```